modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx

```diff
@@ -322,12 +353,24 @@ const RequiredInformation = ({
 		}
 	};
 
-	const handleDescription = () => {
-		if (hasKeyComplimentary) {
-			values.description = purposeDescription;
+	const HandleButtonValue = () => {
+		if (isRenew) {
+			return i18n.sub('renew-x-key', [state?.activationKeys.length]);
+		}
 
-			return values.description;
+		if (infoSelectedKey?.licenseEntryType.includes('Virtual Cluster')) {
```

Comment:
The two variables are constructed differently, one comes from the `state` variable using a `useLocation` and the other comes from the `getNewGenerateKeyFormValues` request. I don't think it's ideal to make this change in this task, it might take a while.

Reply:
ok, we can worry about it later then. thanks